lib/ReactViewModels/ViewState.js

```diff
@@ -122,6 +127,28 @@ export default class ViewState {
       }
     });
 
+    // Reflect preview id from terria when loaded from an init source
+
+    this._sharedFromExplorerPanelSubscription = knockout
+      .getObservable(terria, "sharedFromExplorerPanel")
+      .subscribe(sharedFromExplorerPanel => {
+        if (defined(sharedFromExplorerPanel)) {
+          this.explorerPanelIsVisible = true;
+        }
+      }, this);
+
+    this._previewedItemIdSubscription = knockout
+      .getObservable(terria, "previewedItemId")
+      .subscribe(previewedItemId => {
+        if (
+          defined(
+            previewedItemId && terria.catalog.shareKeyIndex[previewedItemId]
```

Comment:
It may be equivalent in this case, but this code is unusual. I'd think you would either want `if (defined(previewedItemId) && defined(terria.catalog.shareKeyIndex[previewedItemId]))` or `if (previewedItemId && terria.catalog.shareKeyIndex[previewedItemId])`. Using `defined` on the result of a logical `and` is not typically done.

Reply:
Oops this certainly wasn't intended & was going for your latter, I just left off a closing paren.